doc/stp/MSTP_HLD.md

```diff
@@ -0,0 +1,820 @@
+
+Please note that the Edge Port functionality is equivalent to Portfast, with the only difference being the terminology and command usage based on the spanning-tree mode. The configuration for Edge ports will be stored in the portfast attribute in the database and the backend functionality remains the same as portfast.
+
+## Uplink Fast
+MSTP standard does not support uplink fast so uplink fast functionality will be disable for MSTP.
+
+
+#  Sequence Diagrams
+## MSTP global enable
+
+Only the VLANs that are currently present will be mapped to IST instance. A VLAN cannot be mapped to an instance if it has not been created yet.
+
+![MSTP Global Enable](images/MSTP_global_enable.png)
+
+## MSTP global disable
+![MSTP Global Disable](images/MSTP_global_disable.png)
+
+## MSTP region name/version change
+![MSTP Region Config](images/MSTP_region_config.png)
+
+## Instance creation
+![MSTP Instance Create](images/MSTP_instance_create.png)
+
+## Instance deletion
+![MSTP Instance Del](images/MSTP_instance_del.png)
+
+## Add VLAN to instance
+![MSTP VLAN Add](images/MSTP_vlan_add.png)
+
+## Del VLAN from instance
+![MSTP VLAN Del](images/MSTP_vlan_del.png)
+
+## Add VLAN member
+![MSTP VLAN Member Add](images/MSTP_vlan_member_add.png)
+
+## Del VLAN member
+![MSTP VLAN Member Del](images/MSTP_vlan_member_del.png)
+
+Update port-state, topology change and instance-interface events remain the same as depicted in [Sequence Diagrams of PVST](https://github.com/sandeep-kulambi/SONiC/blob/631ab18211e7e396b138ace561b7a04e7f7b49a1/doc/stp/SONiC_PVST_HLD.md#4-flow-diagrams).
+
+# Configuration Commands
+Following configuration commands will be provided for configuration of MSTP:
+## Global Level
+- **config spanning_tree {enable|disable} {mstp|pvst}**
+  - Enables or disables mstp at global level on all ports of the switch.
+  - Only one mode of STP can be enabled at a time.
+- **config spanning_tree max_hops \<max-hops-value\>**
+  - Specify the number of maximum hops before the BPDU is discarded inside a region.
+  - max-hops-value: Default: 20, range: 1-255
+
+## Region Level
+Below commands allow configuring on region basis:
+
+- **config spanning_tree region name \<region-name\>**
+  - Edit the name of region
+  - region-name: Case sensitive, characters should be less than or equal to 32, default: mac-address of bridge
+- **config spanning-tree region revision \<revision-number\>**
+  - Revision number is used to track changes in the configuration and to synchronize the configuration across the switches in the same region.
+  - revision-number: Default: 0, range: 0-65535
+
+## Instance Level
+
+Below commands allow configuration of an instance:
+
+- **config spanning_tree instance (add|del) \<instance-id\>**
+  - Creation or deletion of an instance.
+  - instance-id: Default: 0, range: 1-63
+  - Instance can not be deleted if VLAN(s) are associated with it. 
```

Comment:
The restriction here on an MSTP instance when VLANs are associated with it serves to prevent accidental deletions that could lead to network inconsistencies and loops in VLAN topologies. This approach aligns with industry standards followed by Cisco, Huawei, and SONiC, ensuring consistency and clarity for users familiar with these platforms. Additionally, in SONiC, an interface associated with a VLAN also prevents the deletion of that VLAN, maintaining the integrity of the network configuration.